source/com.microsoft.tfs.core/src/com/microsoft/tfs/core/credentials/internal/GnomeKeyringCredentialsManager.java

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See License.txt in the repository root.
+
+package com.microsoft.tfs.core.credentials.internal;
+
+import java.net.URI;
+
+import com.microsoft.alm.secret.Credential;
+import com.microsoft.alm.secret.Secret;
+import com.microsoft.alm.storage.posix.GnomeKeyringBackedCredentialStore;
+import com.microsoft.tfs.core.Messages;
+import com.microsoft.tfs.core.credentials.CachedCredentials;
+import com.microsoft.tfs.core.credentials.CredentialsManager;
+import com.microsoft.tfs.core.util.URIUtils;
+import com.microsoft.tfs.util.Check;
+import com.microsoft.tfs.util.StringUtil;
+
+public class GnomeKeyringCredentialsManager implements CredentialsManager {
+
+    private final GnomeKeyringBackedCredentialStore gnomeKeyringStore;
+
+    public static boolean isGnomeKeyringSupported() {
+        return GnomeKeyringBackedCredentialStore.isGnomeKeyringSupported();
```

Comment:
The first time is expensive, we need to load the native library libgnome-keyring.so.  The subsequent calls are not as expensive, but we still issue a few calls to gnome-keyring to verify it supports the API we need.

Since we need to make this check anyway, I moved it to a static field as discussed.  This way it's only initialized once during class loading time.